systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java

```diff
@@ -4,28 +4,254 @@
+import io.strimzi.api.kafka.model.balancing.BrokerCapacity;
 import io.strimzi.systemtest.resources.KubernetesResource;
 import io.strimzi.systemtest.resources.ResourceManager;
 import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.resources.crd.KafkaTopicResource;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.StatefulSetUtils;
 import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import io.vertx.core.json.JsonObject;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+import static io.strimzi.api.kafka.model.KafkaResources.kafkaStatefulSetName;
 import static io.strimzi.systemtest.Constants.REGRESSION;
 import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.CoreMatchers.nullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.not;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
 
 @Tag(REGRESSION)
 public class CruiseControlST extends BaseST {
 
     private static final Logger LOGGER = LogManager.getLogger(CruiseControlST.class);
+    private static final String NAMESPACE = "cruise-control-test";
+
+    private static final String CRUISE_CONTROL_NAME = "Cruise Control";
+    private static final int CRUISE_CONTROL_DEFAULT_PORT = 9090;
+    private static final String CRUISE_CONTROL_ENDPOINT = "/kafkacruisecontrol/state";
+    private static final String CRUISE_CONTROL_POD_PREFIX = CLUSTER_NAME + "-cruise-control-";
+
+    private static final String CRUISE_CONTROL_CAPACITY_FILE_PATH = "/tmp/capacity.json";
+    private static final String CRUISE_CONTROL_CONFIGURATION_FILE_PATH = "/tmp/cruisecontrol.properties";
+
+    private static final String CRUISE_CONTROL_METRICS_TOPIC = "strimzi.cruisecontrol.metrics"; // partitions 1 , rf - 1
+    private static final String CRUISE_CONTROL_MODEL_TRAINING_SAMPLES_TOPIC = "strimzi.cruisecontrol.modeltrainingsamples"; // partitions 32 , rf - 2
+    private static final String CRUISE_CONTROL_PARTITION_METRICS_SAMPLES_TOPIC = "strimzi.cruisecontrol.partitionmetricsamples"; // partitions 32 , rf - 2
+
+    @Test
+    void testCruiseControlDeployment()  {
+        String ccStatusCommand = "curl -X GET localhost:" + CRUISE_CONTROL_DEFAULT_PORT + CRUISE_CONTROL_ENDPOINT;
```

Comment:
So we have a java client that is set up to query the state endpoint. Obviously we don;t want to pull the whole `cluster-operator` module into `system-test` but would it make sense to move the client into `operator-common` so that you could use it here? 

WDYT @ppatierno and @kyguy?

Reply:
@tomncooper +1 We can update right after as the java client gets into master

Reply:
Well it may not be worth it for one method but if it turns out we need it for more tests we could move it to `operator-common`

Reply:
The difference between using curl and our Java API client is that the returned JSON could be different.
For example, related to the user tasks endpoint, the client manipulate the original JSON got from the HTTP request in order to produce a simplified one for the specific usage of the operator. Maybe we should avoid this manipulation and moving the logic to extract right information at the operator level? I am not sure that using a client which doesn't return the raw JSON is the right thing to do for the STs.

Reply:
So, do you suggest to use the `curl` utility for all API calls? In that case, we will not be verifying the API itself on the system level, which invokes me another question. It is good approach to use the curl and not API?